sys/shell/shell.c

```diff
@@ -356,7 +362,7 @@ __attribute__((weak)) void shell_post_command_hook(int ret, int argc,
     (void)argc;
 }
 
-static inline void print_prompt(void)
+void print_prompt(void) /* needed externally by module shell_lock */
```

Comment:
This only prints two characters, I don't think you need an external function for that.
I also think we should use a different prompt character for the password prompt, so no need to re-use that function.